pkg/apis/v1alpha1/networkinfo_types.go

```diff
@@ -0,0 +1,56 @@
+/* Copyright © 2024 VMware, Inc. All Rights Reserved.
+   SPDX-License-Identifier: Apache-2.0 */
+
+package v1alpha1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// +genclient
+//+kubebuilder:object:root=true
+//+kubebuilder:storageversion
+
+// NetworkInfo defines network infos for a namespace.
+type NetworkInfo struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	Spec   NetworkInfoSpec   `json:"spec,omitempty"`
+	Status NetworkInfoStatus `json:"status,omitempty"`
+}
+
+//+kubebuilder:object:root=true
+
+// NetworkInfoList contains a list of NetworkInfo.
+type NetworkInfoList struct {
+	metav1.TypeMeta `json:",inline"`
+	metav1.ListMeta `json:"metadata,omitempty"`
+	Items           []NetworkInfo `json:"items"`
+}
+
+// NetworkInfoSpec defines VPC configuration.
+type NetworkInfoSpec struct {
+}
+
+// NetworkInfoStatus defines the observed state of VPC.
+type NetworkInfoStatus struct {
+	VPCs []VPCInfo `json:"vpcs"`
+}
+
+// VPCInfo defines information for VPC.
+type VPCInfo struct {
+	// VPC name.
+	Name       string      `json:"name"`
+	Conditions []Condition `json:"conditions"`
```

Comment:
Do we need to add condition here? or outside of VPCInfo as a general status for NetworkInfo

Reply:
I am thinking the condition here is actually the realization of NSX VPC status.
Since NetworkInfo CR might contain multiple VPCs, they have different condition based on realization status.

Reply:
Questions:
When a VPC is not ready, could we just do not add it to NetworkInfo, or even do not create NetworkInfo?
Today do we have a consumer of the realization status?

Reply:
Yea, I am also thinking if we can remove conditions for now.  Currently when WCPSVC configures cpVM network, it will wait for if VPC is realized successfully before creating SubnetPort CR， I am thinking instead of checking condition status, change to check if there is value for vpcPath in NetworkInfo CR.  WDYT?

Reply:
if a NSX VPC is not realized, then we do not add it to NetworkInfo.

Reply:
That sounds good to me.